### PR TITLE
VL_UTable-row-inprovments_Dmtyro-Holdobin

### PR DESCRIPTION
**Pull request by Explicit12**
- It's possible to pass cell props to nested rows now.
- Prevented onClickRow event when toggle row visibility icon clicked.
- Added warning when row does not have unique id.

### CODE REVIEW

File: src/ui.data-table/UTable.vue [hunk 1]
```diff
@@ -557,6 +558,16 @@ watch(
     tableRows.value = tableRows.value.map((row) => syncRowCheck(row, selectedRows.value));
   },
 );
+watch(
+  () => props.rows,
+  () => {
+    if (!rowsHasId(props.rows)) {
+      // eslint-disable-next-line no-console
+      console.warn("[Vueless][UTable]: Each table row must have unique id.");
```
Comment (itsJohnnyGrid) [inline src/ui.data-table/UTable.vue:566]: maybe it's better to assign the `id` by yourself if it not passed by user. There are a lot of cases where `id` no need to add.

File: src/ui.data-table/UTableRow.vue [hunk 2]
```diff
@@ -78,7 +78,15 @@
...
     @toggle-row-visibility="onClickToggleRowChild"
     @click="onClick"
-  />
+  >
+    <template
+      v-for="(value, key, index) in getFilteredRow(row, columns)"
+      :key="index"
+      #[`cell-${key}`]="slotValues"
+    >
+      <slot :name="`cell-${key}`" :value="slotValues.value" :row="slotValues.row" />
```
Comment (itsJohnnyGrid) [inline src/ui.data-table/UTableRow.vue:87]: what about index as a slot prop?

File: src/ui.data-table/UTableRow.vue [hunk 3]
```diff
@@ -95,7 +103,15 @@
...
         @toggle-row-visibility="onClickToggleRowChild"
         @click="onClick"
-      />
+      >
+        <template
+          v-for="(value, key, index) in getFilteredRow(row, columns)"
+          :key="index"
+          #[`cell-${key}`]="slotValues"
+        >
+          <slot :name="`cell-${key}`" :value="slotValues.value" :row="slotValues.row" />
```
Comment (itsJohnnyGrid) [inline src/ui.data-table/UTableRow.vue:112]: what about index as a slot prop?

File: src/ui.data-table/UTable.vue (hunk 1 above)
Comment (Explicit12) [reply src/ui.data-table/UTable.vue:566]: We should consider this and work with keys in general. I still suggest to move it into once reserved property. Let's leave warning for now and move it to another task for discussion.

File: src/ui.data-table/UTable.vue [hunk 4]
```diff
@@ -196,7 +196,12 @@
                   @binding {string} value
                   @binding {object} row
                 -->
```
Comment (itsJohnnyGrid) [inline src/ui.data-table/UTable.vue:198]: keep in mind to updated slot docs too

**Review by itsJohnnyGrid — APPROVED** [review]

**Pull request merged**